front/src/applications/stdcm/hooks/useLinkedPathSearch.ts

```diff
@@ -0,0 +1,163 @@
+import { useMemo, useState, useCallback } from 'react';
+
+import { compact, groupBy } from 'lodash';
+import nextId from 'react-id-generator';
+import { useSelector } from 'react-redux';
+
+import type { PathItem, SearchResultItemTrainSchedule } from 'common/api/osrdEditoastApi';
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { useOsrdConfActions, useOsrdConfSelectors } from 'common/osrdContext';
+import type { ExtremityPathStepType } from 'common/types';
+import type { StdcmConfSliceActions } from 'reducers/osrdconf/stdcmConf';
+import { useAppDispatch } from 'store';
+import { isEqualDate } from 'utils/date';
+
+import { ArrivalTimeTypes, type StdcmLinkedPathResult, type StdcmLinkedPathStep } from '../types';
+import computeOpSchedules from '../utils/computeOpSchedules';
+
+const useLinkedPathSearch = () => {
+  const dispatch = useAppDispatch();
+  const [postSearch] = osrdEditoastApi.endpoints.postSearch.useMutation();
+
+  const { getTimetableID, getSearchDatetimeWindow } = useOsrdConfSelectors();
+  const { updateOrigin, updateDestination, updateAnteriorPath, updatePosteriorPath } =
```

Comment:
You need to use the new action `updateStdcmPathStep`
```suggestion
  const { updateStdcmPathStep, updateAnteriorPath, updatePosteriorPath } =
```